src/islenska/bincompress.py

```diff
@@ -719,32 +645,45 @@ def make_target(b: str) -> str:
             if inflection_filter is not None and not inflection_filter(beyging):
                 # The user-defined filter fails
                 continue
-            target_beyging = make_target(beyging)
-            if any(t not in target_beyging for t in to_inflection_list if t != "nogr"):
-                # This target beyging string does not contain
-                # our desired variants and is therefore not relevant
-                continue
+            b_set.update(mark_to_set(beyging))
             for form_latin in self.lemma_forms(bin_id):
                 for form_id, mix, kix in self._raw_lookup(form_latin):
                     if form_id != bin_id:
                         continue
                     # Found a word form of the same lemma
                     _, this_beyging = self.meaning(mix)
-                    if this_beyging == target_beyging:
-                        # Found a word form with the target beyging string
+                    if inflection_filter is not None and not inflection_filter(
+                        this_beyging
+                    ):
+                        # The user-defined filter fails
+                        continue
+                    tb_set = mark_to_set(this_beyging)
+                    if not_definite and "gr" in tb_set:
+                        # Asked for no definite form but this form is definite
+                        continue
+                    if tb_set.issuperset(to_inflection):
+                        # Found a word form with the target inflections
                         ksnid_string = self.ksnid_string(kix)
-                        result.add(
-                            Ksnid.from_parameters(
-                                stofn,
-                                bin_id,
-                                ofl,
-                                fl,
-                                form_latin.decode("latin-1"),
-                                this_beyging,
-                                ksnid_string,
-                            )
+                        ks = Ksnid.from_parameters(
+                            stofn,
+                            bin_id,
+                            ofl,
+                            fl,
+                            form_latin.decode("latin-1"),
+                            this_beyging,
+                            ksnid_string,
                         )
-        return result
+                        if ks not in (t[0] for t in results):
```

Comment:
Better?
`if not any(ks == t[0] for t in results):`

Reply:
...or `if all (ks != t[0] for t in results):`

Reply:
Good catch, fixed!